Chap_API_Sharing_Basics.tex

```diff
@@ -0,0 +1,630 @@
+The key can be assigned a new value, but cannot be removed entirely.
+Although key values can originate from different sources, they are always retrieved using the
+\refapi{PMIx_Get} and \refapi{PMIx_Get_nb} API.
+
+The publish/lookup data realm is accessed through a separate set of API's.
+\refapi{PMIx_Put} cannot add or modify key-values within the publish/lookup realm and 
+\refapi{PMIx_Lookup} cannot access key-values outside the publish/lookup realm.
+This data realm is described in detail is chapter [Reference to publish/lookup chapter].
+Although PMIx_Publish and PMIx_Lookup are analogous to PMIx_Put and PMIx_Get in that both pairs 
+of API's provide functionality for exposing and retreiving key-values, the semantics vary significantly.  
+For example, PMIx_Lookup includes the ability to wait until a key is defined before returning.
+PMIx_Publish can restrict the scope of who can access data to its own namespace, the host environment, the session it is executing in, all processes or even custom scopes, while  
+PMIx_Put can restrict the scope of who can access data to the node it is executing on, remote nodes, 
+all nodes or only itself.
+The publish/lookup data realm is useful for advertising information that is not necessarily specific to one process to other processes in the \ac{PMIx} universe.
+The process accessing this information does \emph{not} need to know the identity of the process that provided the data.
+
+\ac{PMIx} does not provide a mechanism to asynchronously notify a process about the availability of key-value information once it is made available by another process.
+However, the nonblocking accessor interfaces (e.g., \refapi{PMIx_Get_nb}, \refapi{PMIx_Lookup_nb}) may provide a degree of asynchronous notification on information availability.
+
+Process related key-value exchanges allow a \ac{PMIx} process to share information specific to 
+itself, and access information specific to one or more processes in the \ac{PMIx} universe.
+These interactions occur within the process data realm.
+The 'put/commit/get' exchange pattern is often used to exchange process related information.
+Optionally, a 'put/commit/fence/get' exchange pattern adds the 'fence' synchronization 
+(and possible collective exchange) for applications that desire it.
+Commonly, these exchange patterns are used in a 
+\declareterm{business card exchange} (a.k.a. \declareterm{modex exchange}) 
+where one \ac{PMIx} client shares its connectivity information, then other \ac{PMIx} clients access that information to establish a connection with that client.
+In some environments that support ``instant-on'' all connectivity information for \ac{PMIx} clients is stored in the job-level information at process creation time and is accessible to the clients without the need to perform any additional key-value exchange.
+
+Keys can exist in multiple data realms, possibly with different values. 
```

Comment:
This feels redundant to prior sections. Perhaps I'm missing the point of this chapter - how is what is presented different from what is in the reserved and non-reserved sections?

Reply:
I have some cleanup to do and we are going to divide this up into 3 PRs that will make this clearer.  The file API_nonreserved_keys.txt is going away (or being renamed to Chap_API_Sharing_Basics.tex.  So this is only presented once.  When I repost this I will use the correct git mv to move it and make it clearer.